modules/public_ip/outputs.tf

```diff
@@ -0,0 +1,9 @@
+output "pip_ids" {
+  description = "The identifiers of the created or sourced Public IP Addresses."
+  value       = { for k, v in local.public_ips : k => v.id }
+}
+
+output "ippre_ids" {
+  description = "The identifiers of the created or sourced Public IP Prefixes."
+  value       = { for k, v in local.public_ip_prefixes : k => v.id }
+}
```

Comment:
What do you think about outputting also [ip_prefix](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/public_ip_prefix#ip_prefix)?

Reply:
thank you @acelebanski